packages/turbo-repository/rust/src/lib.rs

```diff
@@ -131,4 +135,58 @@ impl Workspace {
 
         Ok(map)
     }
+
+    /// Given a set of "changed" files, returns a set of packages that are
+    /// "affected" by the changes.
+    #[napi]
+    pub async fn affected_packages(&self, files: Vec<String>) -> Result<Vec<Package>, Error> {
+        let workspace_root = match AbsoluteSystemPath::new(&self.absolute_path) {
+            Ok(path) => path,
+            Err(e) => return Err(Error::from_reason(e.to_string())),
+        };
+
+        let hash_set_of_paths: HashSet<AnchoredSystemPathBuf> = files
+            .into_iter()
+            .filter_map(|path| {
+                let path_components = path.split(std::path::MAIN_SEPARATOR).collect::<Vec<&str>>();
+                let absolute_path = workspace_root.join_components(&path_components);
+                match workspace_root.anchor(&absolute_path) {
+                    Ok(path) => Some(path),
+                    Err(_) => None,
+                }
+            })
+            .collect();
+
+        // Create a ChangeMapper with no custom global deps or ignore patterns
+        let mapper = ChangeMapper::new(&self.graph, vec![], vec![]);
+        let package_changes = match mapper.changed_packages(hash_set_of_paths, None) {
+            Ok(changes) => changes,
+            Err(e) => return Err(Error::from_reason(e.to_string())),
+        };
+
+        let packages = match package_changes {
+            PackageChanges::All => self
+                .graph
+                .workspaces()
+                .map(|(name, info)| WorkspacePackage {
+                    name: name.to_owned(),
+                    path: info.package_path().to_owned(),
+                })
+                .collect::<Vec<WorkspacePackage>>(),
+            PackageChanges::Some(packages) => packages.into_iter().collect(),
+        };
+
+        let mut serializable_packages: Vec<Package> = packages
+            .into_iter()
+            .map(|p| {
```

Comment:
I think it makes sense to do the filtering before we perform the mapping.
(The second arm shouldn't get hit unless we do something wrong in the package graph, but better safe than sorry.
```suggestion
            .filter(|p| match &p.name {
                WorkspaceName::Root => false,
                WorkspaceName::Other(name) => name != ROOT_PKG_NAME,
            })
            .map(|p| {
```